frame/staking/src/tests.rs

```diff
@@ -5287,3 +5292,160 @@ fn proportional_ledger_slash_works() {
 		BTreeMap::from([(4, 0), (5, value_slashed), (6, 0), (7, 0)])
 	);
 }
+
+#[test]
+fn pre_bonding_era_cannot_be_claimed() {
+	// Verifies initial conditions of mock
```

Comment:
I am not following it strictly myself, but I suggest this about how to document/structure tests: 

https://www.agilealliance.org/glossary/gwt/#:~:text=An%20example%3A,complete%20without%20errors%20or%20warnings

Reply:
Its a little bit difficult with such long tests. Right now its a bit more like given, when1, then1, when2, then2. To strictly follow this we should have every test asserting on only one set of actions (then) and repeat the then1 actions in the `given` for the following test.

Reply:
consider it a suggestion for now then. Yeah, kinda hard to follow in any no-trivial test.